packages/components/pie-notification/test/visual/pie-notification.spec.ts

```diff
@@ -120,201 +42,204 @@ test.describe('Props', () => {
 
     test.describe('PieNotification headingLevels', () => {
         headingLevels.forEach((headingLevel) => {
-            test(`should render correctly the ${headingLevel} headingLevel`, async ({ page, mount }) => {
-                await mount(PieNotification, {
-                    props: {
-                        ...initialValues,
-                        headingLevel,
-                    } as NotificationProps,
-                    slots: { default: slotContent },
-                });
+            test(`should render correctly the ${headingLevel} headingLevel`, async ({ page }) => {
+                // Arrange
+                const basePage = new BasePage(page, 'notification');
+                const props: Partial<NotificationProps> = {
+                    ...initialValues,
+                    headingLevel,
+                };
+
+                await basePage.load({ ...props });
 
+                // Assert
                 await percySnapshot(page, `PieNotification - headingLevel = ${headingLevel} but font size remains the same`);
             });
         });
     });
 
     test.describe('Action buttons', () => {
         test.describe('leadingAction', () => {
-            test('should render leadingAction when leadingAction is provided', async ({ mount, page }) => {
+            test('should render leadingAction when leadingAction is provided', async ({ page }) => {
                 // Arrange
-                await page.setViewportSize({ width: 375, height: 667 });
-                await mount(PieNotification, {
-                    props: {
-                        ...initialValues,
-                        leadingAction: mainAction,
-                    } as NotificationProps,
-                    slots: { default: slotContent },
-                });
+                const basePage = new BasePage(page, 'notification');
+                const props: Partial<NotificationProps> = {
+                    ...initialValues,
+                    leadingAction: mainAction,
+                };
```

Comment:
I'm unable to comment on the associated line, but could you rename `mainAction` to `leadingAction` then you can do:

```suggestion
                const props: Partial<NotificationProps> = {
                    ...initialValues,
                    leadingAction,
                };
```

Reply:
Turns out I can just remove the whole leading action thing completely as its already pre-defined in the `initialValues` so I'll remove them all where its not required unless the test specifies otherwise.